src/main/java/networkbook/storage/JsonAdaptedPerson.java

```diff
@@ -193,6 +197,14 @@ private UniqueList<Course> getModelCourses() throws IllegalValueException, Dupli
         return modelCourses;
     }
 
+    private Course getModelCourse(JsonAdaptedProperty<Course> jsonCourse) throws IllegalValueException {
+        try {
+            return ParserUtil.parseCourseWithPrefixes(jsonCourse.getName());
+        } catch (ParseException e) {
+            throw new IllegalValueException(Course.MESSAGE_CONSTRAINTS);
```

Comment:
If user changes the storage such that end date is before the start date, the error message is misleading.
But you can decide whether you want to merge first and fix this later, or fix now,
since message only appears in the log.

Reply:
Ok, I'll create a new issue for this then